### ISSUE TITLE
Improved tag handling by updating keys and suggesting values

### ISSUE THREAD

**Issue by MiMoHo** [osmandapp/OsmAnd#18654]
### Describe the idea (required)

For advanced tags, I noticed several [keys](https://wiki.openstreetmap.org/wiki/Key) miss categories like "[commerce](https://wiki.openstreetmap.org/wiki/Category:Commerce)", [bridge](https://wiki.openstreetmap.org/wiki/Key:bridge) types, [language](https://wiki.openstreetmap.org/wiki/Key:language:*) and [indoor_seating](https://wiki.openstreetmap.org/wiki/Key:indoor_seating) or values like "[diet:meat](https://wiki.openstreetmap.org/wiki/Key:diet:*)" or "[charge_point](https://wiki.openstreetmap.org/wiki/Tag:man_made%3Dcharge_point)".

Possible values do not show up when tapping into the _Advanced_ field unless typing at least three characters. When the user doesn't have an idea what to search for, it can be tricky to find the desired term assuming it is actually registered in the app.

Some values are likely to be entered twice, e.g. when entering the website name in the _Basic_ information which results in the tag "website", but when searching for "website" in _Advanced_, the only suggested option is "contact:website".

Opening hours do not allow for special information and formats like (certain) public holidays #11344 or seasonal opening hours under _Basic_. OsmAnd doesn't assist to enter the right format even in _Advanced_ to solve issues like #7857 and #15072. This can result in showing multiple untitled entries of opening hours in editing mode that were entered in another app as reported in #11390 or false seasonal 24/7 info: #11571

### Tell us about the expected behaviour (required)

Compare keys and values from [OSM Wiki](https://wiki.openstreetmap.org) with the current options in OsmAnd and update the app's keys and values.

### Tell us about alternatives you've considered (required)

Entering more values myself but was restricted as I "must be on a branch".

**Comment by scaidermern** [comment]
I usually avoid the internal OSM editor because of its limited interface, broken suggestion system and non-existing validations. Any other mobile editor (EveryDoor, Vespucci, StreetComple) is way more useful and less error-prone.

Just a hint, not solving any of the above issues.

(I would even drop OsmAnd's editor completely 🤷‍♂️)

**Comment by pebogufi** [comment]
No. Don't remove internal editor. I also use Vespucci quite often, but for **simple** editing tasks the use of internal editor is faster than changing to Vespucci each time.

**Comment by vshcherb** [comment]
OsmAnd only supports only tag that it are embedded i.e. icons / translations. 
So yes it's limited but on other hand it also shows what's supported. In case something is missing, it's a good indication that we need to add that tag support. 

On the other hand - Templates and Better UX is essential and that what we need to add support and focus.

Also I believe we need to drop OSM Editing Plugin cause it could be nicer integrated into app itself to invite bigger audience.

**Comment by rene78** [comment]
OsmAnd is my go-to app for everything related to navigation and OSM mapping. Advanced editing features similarly to the ones found in apps listed by @scaidermern would be amazing!

**Comment by MiMoHo** [comment]
Following up with some data to make this actionable, building on @vshcherb's point that missing tags are "a good indication that we need to add that tag support".

I compared the officially documented OSM Map Features against `poi_types.xml` (the registry driving the editor suggestions):

**Methodology**
- "Officially documented" = English wiki tag pages with status *approved* / *de facto* / *in use*, mappable on nodes or areas (source: taginfo wiki database, July 2026).
- Compared against all `(tag, value)` pairs in `resources/poi/poi_types.xml` including `tag2`/`edit_tag` (4,266 pairs today).
- Documented values below 1,000 worldwide uses and infrastructure micro-mapping (fences, signals, utility poles, …) are excluded from the proposal for now.

**Results** (full report: https://gist.github.com/MiMoHo/4a6361ff5fbb5d1559e55f65c1be45be)

| Group | Values |
|---|---|
| A. Documented, genuinely missing, ≥ 1,000 uses | **184** (man_made 49, amenity 29, shop 16, natural 16, sport 13, historic 12, water 12, …) |
| B. Divergences — documented under a key OsmAnd maps differently (e.g. `healthcare=pharmacy` vs. OsmAnd's `amenity=pharmacy`, `water=reservoir` vs. `landuse=reservoir`) | **80** |
| C. Excluded for now (low usage: 413, infrastructure micro-mapping: 113) | 526 |

Top gaps from group A by usage: `emergency=access_point` (195k, in use), `healthcare=doctor` (150k, **approved**), `shop=storage_rental` (39k, de facto), `amenity=loading_dock` (39k, approved), `sport=pickleball` (29k), `historic=citywalls` (17k), `sport=padel` (17k), `office=coworking` (5.7k), …

**Proposal**
1. Curated batch PRs to OsmAnd-resources, one theme at a time (e.g. healthcare, sport, shop), each ~20–40 values: `poi_type` entries + English `phrases` keys; icons fall back to the category icon and can follow later.
2. For group B, align with the documented scheme where sensible — e.g. accept the documented key as an edit alternative (`edit_tag`) so both the OsmAnd mapping and the official tagging are recognized in the editor.
3. Keep group C out of scope until there is real usage.

Would you take such PRs? Any preference on batch size or where to start?

*Analysis prepared with Claude Code (Claude Fable 5, xhigh).*
